msteams-platform/bots/streaming-ux.md

```diff
@@ -0,0 +1,586 @@
+    - **Enable AI-powered features for final message**: Set attachments in the final chunk using the `EndStreamHandler` within the `ActionPlanner` declaration.
+
+The following code snippet shows an example of streaming bot messages:
+
+# [C#](#tab/csharp)
+
+```C#
+    // Create OpenAI Model
+    builder.Services.AddSingleton<OpenAIModel > (sp => new(
+        new OpenAIModelOptions(config.OpenAI.ApiKey, "gpt-4o")
+        {
+            LogRequests = true,
+            Stream = true,              // Set stream toggle
+        },
+        sp.GetService<ILoggerFactory>()
+    ));
+
+ResponseReceivedHandler endStreamHandler = new((object sender, ResponseReceivedEventArgs args) =>
+    {
+        StreamingResponse? streamer = args.Streamer;
+
+        if (streamer == null)
+        {
+            return;
+        }
+
+        AdaptiveCard adaptiveCard = new("1.6")
+        {
+            Body = [new AdaptiveTextBlock(streamer.Message) { Wrap = true }]
+        };
+
+        var adaptiveCardAttachment = new Attachment()
+        {
+            ContentType = "application/vnd.microsoft.card.adaptive",
+            Content = adaptiveCard,
+        };
+
+
+        streamer.Attachments = [adaptiveCardAttachment];    // Set attachments
+
+    });
+
+
+    // Create ActionPlanner
+    ActionPlanner<TurnState> planner = new(
+        options: new(
+            model: sp.GetService<OpenAIModel>()!,
+            prompts: prompts,
+            defaultPrompt: async (context, state, planner) =>
+            {
+                PromptTemplate template = prompts.GetPrompt("Chat");
+                return await Task.FromResult(template);
+            }
+        )
+        {
+            LogRepairs = true,
+            StartStreamingMessage = "Loading stream results...", // Set informative message
+            EndStreamHandler = endStreamHandler // Set final chunk handler
+        },
+        loggerFactory: loggerFactory
+    );
+```
+
+# [JavaScript or TypeScript](#tab/jsts)
+
+```JavaScript
+const model = new OpenAIModel({
+    // ...Setup OpenAI or AzureOpenAI
+    stream: true,                                         // Set stream toggle
+});
+
+const endStreamHandler: PromptCompletionModelResponseReceivedEvent = (ctx, memory, response, streamer) => {
+    // ... Setup attachments
+    streamer.setAttachments([...cards]);                      // Set attachments
+};
+
+const planner = new ActionPlanner({
+    model,
+    prompts,
+    defaultPrompt: 'default',
+    startStreamingMessage: 'Loading stream results...', // Set informative message
+    endStreamHandler: endStreamHandler                  // Set final chunk handler
+});
+```
+
+# [Python](#tab/python)
+
+```Python
+model = OpenAIModel(
+        OpenAIModelOptions(api_key=config.OPENAI_KEY, default_model="gpt-4o", stream=True)
```

Comment:
Can we add a comment here

`Set stream toggle`

Reply:
@lilyydu We are unable to make changes to code snippets. Please update and provide the code snippet.